alfalfa_worker/worker_fmu/step_fmu.py

```diff
@@ -164,13 +169,41 @@ def run(self):
                         self.set_idle_state()
                         break
         else:
-            while self.simtime < self.endTime:
-                if self.db_stop_set():
-                    break
+            # first step takes extra time and needs to happen outside timescale loop
+            if self.simtime == self.startTime:
+                print("taking first step at ", datetime.now())
                 self.step()
-                # TODO: Make this respect time scale provided by user
-                time.sleep(5)
+                print("finished first step at ", datetime.now())
+                next
+            current_time = datetime.now()
+            next_step_time = current_time + self.realworld_timedelta
+            print("in run with current time & next_step_time: ", current_time, next_step_time)
+            self.advance = False
+            while True:
+                current_time = datetime.now()
+
+                if current_time >= next_step_time:
+                    self.advance = True
+                    # sim is not keeping up with target timescale.
+                    # TODO rethink arbitrary 60s behind
+                    if (current_time > next_step_time + timedelta(seconds=60)):
```

Comment:
In the future we could keep track of step times and stop if step run times are routinely take longer than the target times.

Reply:
yes, we can probably do better.  Also note that this logic should be shared between fmu and osws.